plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -761,6 +762,13 @@ private ObjectMetadata getS3ObjectMetadata(Path path, String bucketName, String
                         }
                         catch (RuntimeException e) {
                             STATS.newGetMetadataError();
+                            // Retry AWS STS Throttling error code specifically since it uses HTTP_BAD_REQUEST status code
```

Comment:
Why do you want to retry on throttling? That may cause throttling more likely for others. Am I right?

Reply:
TLDR; yes, retrying a throttled request does have a chance of causing more throttling errors, but I think retrying these requests using the existing retry logic is overall a net positive since the throttling errors can happen in the middle of the query depending when the session expires.

I can't seem to find any documentation online on their rate limits for AWS STS assumeRole requests (probably for good reason), but our use case utilizes a lot of cross-account Glue catalogs, Trino is only one piece that we need to generate a cross account IAM role session. During times of high concurrency in other parts of our system, we too have to manually handle and retry these throttling errors.

I'm not an expert on how Trino uses S3 credentials for a S3 Hive catalog, but I imagine, once a valid session is generated for a catalog on a node, that node no longer needs to call assumeRole until the session expires.

However, it seems that API calls to glue [use a different credential provider](https://github.com/trinodb/trino/blob/9298c516f9d2e695fc50e9a1684aebe69b555fc5/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java#L232),. It is very rare for us to see it occur during the metastore API calls for our application, I'll have to leave this part for when time permits.